### DISCUSSION TITLE
What I'm doing wrong with proxied server?

### DISCUSSION THREAD

**Discussion by SirLouen** [bluenviron/mediamtx#1713] [Questions]
I tried to go for the simplest option of all

```
docker run --rm -it -e MTX_PROTOCOLS=tcp -e MTX_PATHS_PROXIED_SOURCE=http://honjin1.miemasu.net/nphMotionJpeg -p 8554:8554 -p 1935:1935 -p 8888:8888 -p 8889:8889 aler9/rtsp-simple-server
```

Then I run `rtsp://localhost:8554/proxied` in my RTSP windows viewer (5KPlayer or VLC) and it shows the error 

```
2023/04/16 12:05:42 INF MediaMTX / rtsp-simple-server v0.22.2
2023/04/16 12:05:42 INF [path proxied] [hls source] started
2023/04/16 12:05:42 INF [RTSP] listener opened on :8554 (TCP)
2023/04/16 12:05:42 INF [RTMP] listener opened on :1935
2023/04/16 12:05:42 INF [HLS] listener opened on :8888
2023/04/16 12:05:42 INF [WebRTC] listener opened on :8889 (HTTP)
2023/04/16 12:05:45 INF [RTSP] [conn 172.17.0.1:43484] opened
2023/04/16 12:05:45 INF [RTSP] [conn 172.17.0.1:43484] closed (no one is publishing to path 'proxied')
2023/04/16 12:05:45 INF [RTSP] [conn 172.17.0.1:43486] opened
2023/04/16 12:05:45 INF [RTSP] [session 86b397ef] created by 172.17.0.1:43486
2023/04/16 12:05:45 INF [RTSP] [conn 172.17.0.1:43486] closed (invalid URL ((null)))
2023/04/16 12:05:45 INF [RTSP] [session 86b397ef] destroyed (not in use)
2023/04/16 12:05:45 INF [RTSP] [conn 172.17.0.1:43498] opened
2023/04/16 12:05:45 INF [RTSP] [conn 172.17.0.1:43498] closed (EOF)
```

What I'm doing wrong?

I've also tried the Windows binary straight away with a extremely simple ' mediamtx.yml` file like this

```
api: yes

paths:
  proxied:
    source: http://honjin1.miemasu.net/nphMotionJpeg
```

And still doesnt' work when I go to `rtsp://localhost:8554/proxied`

```
2023/04/16 15:10:25 INF [RTSP] [session 4904d1fa] created by [::1]:23755
2023/04/16 15:10:25 INF [RTSP] [conn [::1]:23755] closed (path of a SETUP request must end with a slash. This typically happens when VLC fails a request, and then switches to an unsupported RTSP dialect)
2023/04/16 15:10:25 INF [RTSP] [session 4904d1fa] destroyed (not in use)
2023/04/16 15:10:25 INF [RTSP] [conn [::1]:23756] opened
2023/04/16 15:10:25 INF [RTSP] [conn [::1]:23756] closed (EOF)
```

API @ 9997 do work by the way like `http://127.0.0.1:9997/v1/config/get`

Also `http://127.0.0.1:8888/proxied/`  returns a 404 page not Found

Which logs:

```
2023/04/16 15:25:31 INF [HLS] [muxer proxied] created (requested by 127.0.0.1)
2023/04/16 15:25:31 INF [HLS] [muxer proxied] destroyed (no one is publishing to path 'proxied')
```

**Comment by aler9** [comment]
Hello ingesting Motion-JPEG streams is something that is not currently supported. Nonetheless, i've created a feature request in order to track the development of the feature: #1714

**Reply by SirLouen** [reply]
@aler9 I was wondering if this could be done through proxied with a ffmpeg command (theorically MJPEG streams can be ingested through ffmpeg afaik?

Also, does this have any relationship with this issue I present here ?
https://github.com/aler9/mediamtx/issues/1368#issuecomment-1509931090? 

Or that HLS is completely different type of stream?

I have to say that I've tried with one RTSP source I have from a local cam in my site, and it's working fine 👍 the proxied service so the problem is definitelly that MJPEG thing I used for testing, but the real stream I'm tying to accoplish is that HLS in m3u8 I comment on that issue I mention above

**Reply by aler9** [reply]
You can certainly use ffmpeg to ingest MJPEG:

```yml
paths:
  cam:
    runOnInit: ffmpeg -i http://... --pix_fmt yuv420p -c:v libx264 -preset ultrafast -b:v 600k -max_muxing_queue_size 1024 -f rtsp rtsp://localhost:$RTSP_PORT/$RTSP_PATH
```

HLS is a completely different kind of stream.

**Reply by SirLouen** [reply]
> You can certainly use ffmpeg to ingest MJPEG:

I used this to make it work because not sure why but your command failed.

```
paths:
  proxied:
    runOnInit: ffmpeg -use_wallclock_as_timestamps 1 -re -i http://honjin1.miemasu.net/nphMotionJpeg -vcodec libx264 -f rtsp rtsp://localhost:$RTSP_PORT/$RTSP_PATH
```

Apart from this, do you know what could be the issue with the HLS stream from the issue https://github.com/aler9/mediamtx/issues/1368#issuecomment-1509931090?

Could be a bug or something?

[a comment by an automated account is not shown]
